module/applications/actor/base-sheet.mjs

```diff
@@ -120,4 +123,11 @@ export default class ActorSheetEd extends ActorSheet {
     const item = this.actor.items.get( li.dataset.itemId );
     return item.sheet?.render( true );
   }
+
+  async _onKarmaRefresh ( ) {
```

Comment:
Just like with the weightCalculation, this should an appropriate method in the actor document, e.g. `this.actor.refreshKarma();`.
The function in the actor document class should like this, using the update method to persist the changes:
```javascript
karmaRitual() {
    this.update( {"system.karma.value": this.system.karma.max} );
}
```

Reply:
ok, moved it as I understood it

Reply:
Did some minor cleanup in commit 455b8f8cc3aee330be00d33c1a2ed8c1c9dc426a. If that is fine, mark this as resolved and we can merge it.